rest_framework_mongoengine/fields.py

```diff
@@ -279,7 +279,7 @@ def to_internal_value(self, value):
             doc_id = self.parse_id(value)
 
         try:
-            return self.get_queryset().only('id').get(id=doc_id).to_dbref()
+            return self.get_queryset().only('pk').get(pk=doc_id).to_dbref()
```

Comment:
When a model has a custom primary key, the `_fields` attribute of its class does not contain `"id"`.